mlflow/sklearn/__init__.py

```diff
@@ -724,16 +724,21 @@ def fetch_logged_data(run_id):
             stacklevel=2,
         )
 
-    def fit_mlflow(self, func_name, *args, **kwargs):
+    def fit_mlflow(self, clazz, func_name, *args, **kwargs):
+        """
+        Autologging function that performs model training by executing the training method
+        referred to be `func_name` on the instance of `clazz` referred to by `self` & records
+        MLflow parameters, metrics, tags, and artifacts to a corresponding MLflow Run.
+        """
         should_start_run = mlflow.active_run() is None
         if should_start_run:
             try_mlflow_log(mlflow.start_run)
 
         _log_pretraining_metadata(self, *args, **kwargs)
 
-        original_fit = gorilla.get_original_attribute(self, func_name)
+        original_fit = gorilla.get_original_attribute(clazz, func_name)
```

Comment:
This (together line 914) is the critical change to address problem (2) from https://github.com/mlflow/mlflow/issues/3574#issuecomment-714994583. 

Assume that the old line was being invoked from a patched `fit` call on a class called `Parent` and that `self` refers to an instance of class `Child` which inherits from `Parent`. Calling `get_original_attribute()` on `self` would yield the `Child.fit()` function, rather than `Parent.fit()`, leading to recursive behavior.

Now, instead of fetching the attribute from `self`, we fetch it from `clazz` (which would refer to `Parent` in the above example).